### PR TITLE
etcdserver: align 64-bit atomics on 8-byte boundary

### PR DESCRIPTION
**Pull request by luxas**
I read through the whole source code today, and came across this and #4733 that didn't obey the rules. 
@xiang90 @heyitsanthony

### CODE REVIEW

**Comment by luxas** [comment]
Code in `Godeps` that uses `atomic`:
 - `Godeps/_workspace/src/github.com/cheggaaa/pb/pb.go`
   - has aligned the `int64` field and commented why
 - `Godeps/_workspace/src/github.com/golang/glog/glog.go`
   - has aligned the fields. common package
 - `Godeps/_workspace/src/github.com/prometheus/client_golang/prometheus/histogram.go`
   - has aligned the fields and commented why
 - `Godeps/_workspace/src/github.com/prometheus/client_golang/prometheus/value.go`
   - has aligned the fields and commented why
 - `Godeps/_workspace/src/github.com/spacejam/loghisto/metrics.go`
   - has only `map`s with `int64`, haven't tested if they need to be aligned as well
 - `Godeps/_workspace/src/golang.org/x/net/trace/events.go`, `Godeps/_workspace/src/golang.org/x/net/trace/trace.go` 
   - hasn't aligned `eventLog.refs int32` and `trace.refs int32` but I don't know if it's necessary. The `net` package should be stable as well, so it shouldn't be a problem.
 - `Godeps/_workspace/src/google.golang.org/grpc/transport/http_util.go`
   - aligned `int32` but no comment.

That was all usage of `atomic` in `Godeps`. With this PR and #4733 all `etcd` code is aligned too, and #4715 fixes compilation of `32-bit`

File: etcdserver/server.go [hunk 1]
```diff
@@ -153,10 +153,14 @@ type Server interface {
...
 // EtcdServer is the production implementation of the Server interface
 type EtcdServer struct {
-	// r must be the first element to keep 64-bit alignment for atomic
+	// r and inflightSnapshots must be the first elements to keep 64-bit alignment for atomic
 	// access to fields
 	r raftNode
 
+	// count the number of inflight snapshots.
+	// MUST use atomic operation to access this field.
+	inflightSnapshots int64
```
Comment (heyitsanthony) [inline etcdserver/server.go]: this should be above raftNode for 8-byte alignment; sizeof(raftNode) == 100 on 386

**Comment by heyitsanthony** [comment]
gofmt, change commit title to `etcdserver: align 64-bit atomics on 8-byte boundary`, and it's lgtm. Thanks!

**Comment by luxas** [comment]
Yep, saw that

**Comment by heyitsanthony** [comment]
commit message still looks off-- `git commit --amend`...? Thanks!

**Comment by heyitsanthony** [comment]
lgtm

**Comment by luxas** [comment]
Ah, you meant commit message too. Fixed

**Comment by luxas** [comment]
@xiang90 This has been lgtm'd, so maybe you could merge?

**Comment by philips** [comment]
@luxas @heyitsanthony Is there any tool we can run to test for this? It is super annoying that we have no way to detect a regression here.

**Comment by heyitsanthony** [comment]
@philips unaware of any tool to check this. A test case using `unsafe.Offsetof` on `EtcdServer` would be feasible but incomplete; it won't catch misuse like `type s struct { x int32; s EtcdServer }`...

**Comment by luxas** [comment]
Sorry, but I do not understand why this hasn't been merged yet.
@xiang90 @hongchaodeng @heyitsanthony

**Comment by philips** [comment]
@luxas It is just really annoying for us to merge this and then have no way of knowing if we break it again. I am going to email go-nuts for help.

**Comment by philips** [comment]
Posted: https://groups.google.com/forum/#!topic/golang-nuts/l5mzkGVmNp0

**Comment by luxas** [comment]
@philips OK, but this fixes the atomic alignment, not breaking it.
#4733 and this one was uploaded less than an hour after each other and this PR initially had the changes that #4733 have. 

What's left is what #4733 didn't catch. The thing is, #4733 was merged less than twenty minutes after upload and this PR has been pending a month now. I just don't understand why, given it has been lgtm'd already

Do not get me wrong, I want to contribute to multiarch etcd, but with this treatment I'm not sure what you want.

Of course, +1 for a integration test that checks all structs for this, if we get one from the thread.

**Comment by luxas** [comment]
@heyitsanthony see here too. This has lgtm but isn't merged.

**Comment by philips** [comment]
@luxas OK, I will let @heyitsanthony make the merge but I filed a bug here to remind us to find a static analysis tool: https://github.com/coreos/etcd/issues/5027

LGTM

**Pull request merged**